applications/StructuralMechanicsApplication/custom_response_functions/response_utilities/element_finite_difference_utility.cpp

```diff
@@ -222,14 +201,11 @@ namespace Kratos
                 rElement.CalculateMassMatrix(perturbed_mass_matrix, rCurrentProcessInfo);
 
                 //compute derivative of RHS w.r.t. design variable with finite differences
-                noalias(rOutput) = (perturbed_mass_matrix - unperturbed_mass_matrix) / rPertubationSize;
+                noalias(rOutput) = (perturbed_mass_matrix - rMassMatrix) / rPertubationSize;
 
                  // unperturb the design variable
                 rNode.GetInitialPosition()[coord_dir] -= rPertubationSize;
                 rNode.Coordinates()[coord_dir] -= rPertubationSize;
-
-                //call one last time to make sure everything is as it was before TODO improve this..
-                rElement.CalculateMassMatrix(perturbed_mass_matrix, rCurrentProcessInfo);
```

Comment:
@MFusseder maybe you have an idea how to test this.
For the ShapeDerivatives it is seems not to be needed.